src/Telegram.Bot/Requests/FileRequestBase.cs

```diff
@@ -1,71 +1,45 @@
-using System.Globalization;
-using System.Linq;
+using System.Text;
 
 namespace Telegram.Bot.Requests;
 
-/// <summary>
-/// Represents an API request with a file
-/// </summary>
+/// <summary>Represents an API request with a file</summary>
 /// <typeparam name="TResponse">Type of result expected in result</typeparam>
-
-public abstract class FileRequestBase<TResponse> : RequestBase<TResponse>
+/// <param name="methodName">Bot API method</param>
+public abstract class FileRequestBase<TResponse>(string methodName) : RequestBase<TResponse>(methodName)
 {
-    /// <summary>
-    /// Initializes an instance of request
-    /// </summary>
-    /// <param name="methodName">Bot API method</param>
-    protected FileRequestBase(string methodName)
-        : base(methodName)
-    { }
-
-    /// <summary>
-    /// Initializes an instance of request
-    /// </summary>
-    /// <param name="methodName">Bot API method</param>
-    /// <param name="method">HTTP method to use</param>
-    protected FileRequestBase(string methodName, HttpMethod method)
-        : base(methodName, method)
-    { }
+    readonly static Encoding Latin1 = Encoding.GetEncoding(28591);
 
-    /// <summary>
-    /// Generate multipart form data content
-    /// </summary>
-    /// <param name="fileParameterName"></param>
-    /// <param name="inputFile"></param>
-    /// <returns></returns>
-    protected MultipartFormDataContent ToMultipartFormDataContent(
-        string fileParameterName,
-        InputFileStream inputFile)
+    /// <inheritdoc/>
+    public override HttpContent? ToHttpContent()
     {
-        if (inputFile is null or { Content: null })
+        InputFileConverter.Attachments.Value = null;
+        var utf8Json = JsonSerializer.SerializeToUtf8Bytes(this, GetType(), JsonBotAPI.Options);
+        var attachments = InputFileConverter.Attachments.Value;
```

Comment:
I'm not sure if this is a good idea to make such a tight coupling between `InputFileConverter` and `FileRequestBase`. It's really hard to follow such code.

Reply:
It doesn't seem so illogical to have a coupling between those, they work toward the same goal. still makes things much simpler, avoid the many ToHttpMethod